Simplify docs/cli/code/index texts (#35183)
hashicorp/terraform: website/docs/cli/code/index.mdx

---
page_title: Format and validate Terraform configuration using the Terraform CLI 

> description: Learn about the Terraform commands that validate, format, and upgrade code written in HCL.

Reviewer: ```suggestion
description: Learn about the Terraform commands that validate, format, and upgrade code written in HCL.
---
```

---
page_title: Writing and Modifying Code - Terraform CLI
description: >-
  Learn commands that help validate, format, and upgrade code written in 

>   Terraform's configuration language.

Author: Used in different places like this, e.g.  

- https://developer.hashicorp.com/terraform/language/style
- https://developer.hashicorp.com/terraform/language/syntax/configuration

Other places use "the Terraform language".

page_title: Writing and Modifying Code - Terraform CLI
description: >-

>   Learn commands that help validate, format, and upgrade code written in the

Author: ```suggestion
  Learn commands that help validate, format, and upgrade code written in 
```

oups, I missed this. 

"Terraform's configuration language" is also used here: https://developer.hashicorp.com/terraform/language/syntax/configuration

  well as a pre-commit hook in your version control system.

- [`terraform validate`](/terraform/cli/commands/validate): Validate the
  syntax and arguments of the Terraform configuration files in a directory,
  including argument and attribute names and types for resources and modules.
  The `plan` and `apply` commands automatically validate a configuration before
  performing any other work. `validate` isn't a crucial part of the core
  workflow but can be very useful as a pre-commit hook or as part of a
  continuous integration pipeline.

- [`0.13upgrade`](/terraform/cli/commands/0.13upgrade) and
  [`0.12upgrade`](/terraform/cli/commands/0.12upgrade): Modify the configuration files in a Terraform module automatically to help deal with major
  syntax changes that occurred in the 0.13 and 0.12 releases of Terraform. Both
  commands are only available in the Terraform version they are
  associated with. Make sure to upgrade older code to be compatible
  with 0.12 before attempting to make it compatible with 0.13. For more detailed
  information about updating code for new Terraform versions, see the [upgrade

>   guides](/terraform/language/upgrade-guides).

Author: IDK the pattern but I would leave it out. If the user is interested and clicks the links they land on the page in the language docs anyway.

Just in case, you wanna revert my change: 

```suggestion
  guides](/terraform/language/upgrade-guides) in the Terraform language docs.
```

> to work with. Integrating these commands into your editing workflow can
> potentially save you time and effort.

Author: ```suggestion
to work with. The following commands can be integrated into your editing workflow and 
potentially save you time and effort:
```

> - [`terraform console`](/terraform/cli/commands/console): Start an 

Author: Review note: 

Here, we use "the" before and "command" after the code formatted command part (as in my other places). Both words are not repeated but "command" is used once for all of them in the end: 
https://developer.hashicorp.com/terraform/cli/workspaces. 

With the intro, this pattern doesn't make sense in my opinion. The following commands simply need a list of commands, so "the" and "command" should not be repeated.

I also thought about referring to the command with an "s", e.g. "starts", but decided to go for directly addressing the user.

Reviewer 2: Thank you for your thought here! To meet our style guide, bullets have to be full sentences, so I suggest tweaking to:
```suggestion
- The [`terraform console`](/terraform/cli/commands/console) command starts an 
```

Reviewer 2: Apologies for the delay in review! In the interest of getting this merged in now, I'm going to add in my style suggestions and merge this in ✨Thank you for making our docs better!

> - [`terraform console`](/terraform/cli/commands/console): Start an
>   interactive shell for evaluating Terraform
>   [expressions](/terraform/language/expressions), which can be a faster way
>   to verify that a particular resource argument results in the value you expect.

Author: ```suggestion
- [`terraform console`](/terraform/cli/commands/console): Start an 
  interactive shell for evaluating Terraform
  [expressions](/terraform/language/expressions) to verify that a 
  particular resource argument results in the value you expect.
```

> - [`terraform fmt`](/terraform/cli/commands/fmt): Rewrite Terraform
>   configuration files to a canonical format and style, so you don't have to
>   waste time making minor adjustments for readability and consistency. It works
>   well as a pre-commit hook in your version control system.

Author: ```suggestion
- [`terraform fmt`](/terraform/cli/commands/fmt): Rewrite Terraform
  configuration files to a canonical format and style, so you don't have to
  waste time making minor adjustments for readability and consistency. `fmt` works
  well as a pre-commit hook in your version control system.
```

to work with. Integrating these commands into your editing workflow can
potentially save you time and effort.

> - [`terraform console`](/terraform/cli/commands/console): Starts an

Author: ```suggestion
- [`terraform console`](/terraform/cli/commands/console): Start an
```

  interactive shell for evaluating Terraform
  [expressions](/terraform/language/expressions), which can be a faster way
  to verify that a particular resource argument results in the value you expect.

> - [`terraform fmt`](/terraform/cli/commands/fmt): Rewrites Terraform

Author: ```suggestion
- [`terraform fmt`](/terraform/cli/commands/fmt): Rewrite Terraform
```

  configuration files to a canonical format and style, so you don't have to
  waste time making minor adjustments for readability and consistency. It works
  well as a pre-commit hook in your version control system.

> - [`terraform validate`](/terraform/cli/commands/validate): Validates the

Author: ```suggestion
- [`terraform validate`](/terraform/cli/commands/validate): Validate the
```

>   syntax and arguments of the Terraform configuration files in a directory,
>   including argument and attribute names and types for resources and modules.
>   The `plan` and `apply` commands automatically validate a configuration before
>   performing any other work, so `validate` isn't a crucial part of the core
>   workflow, but it can be very useful as a pre-commit hook or as part of a
>   continuous integration pipeline.

Author: ```suggestion
  syntax and arguments of the Terraform configuration files in a directory,
  including argument and attribute names and types for resources and modules.
  The `plan` and `apply` commands automatically validate a configuration before
  performing any other work. `validate` isn't a crucial part of the core
  workflow but can be very useful as a pre-commit hook or as part of a
  continuous integration pipeline.
```

- [`0.13upgrade`](/terraform/cli/commands/0.13upgrade) and

>   [`0.12upgrade`](/terraform/cli/commands/0.12upgrade): The commands can automatically
>   modify the configuration files in a Terraform module to help deal with major

Author: ```suggestion
  [`0.12upgrade`](/terraform/cli/commands/0.12upgrade): Modify the configuration files in a Terraform module automatically to help deal with major
```

  syntax changes that occurred in the 0.13 and 0.12 releases of Terraform. Both
  commands are only available in the Terraform version they are

>   associated with. Always upgrade older code to be compatible

Author: ```suggestion
  associated with. Make sure to upgrade older code to be compatible
```

## Workflows 

Terraform CLI includes several commands to make Terraform code more convenient

> to work with. The following commands can be integrated into your editing 

Reviewer 2: ```suggestion
to work with. You can integrate the following commands into your editing 
```

> workflow and may save you time and effort:

Reviewer 2: I know this was in the original, but I'd suggest we make this a little less vague and more direct:
```suggestion
workflow to test expressions and format and validate your configuration syntax:
```

- [`terraform console`](/terraform/cli/commands/console): Start an 
  interactive shell for evaluating Terraform

>   [expressions](/terraform/language/expressions) to quickly verify that a 

Reviewer 2: ```suggestion
  [expressions](/terraform/language/expressions), to quickly verify that a 
```

  particular resource argument results in the value you expect.

> - [`terraform fmt`](/terraform/cli/commands/fmt): Rewrite Terraform

Reviewer 2: ```suggestion
- The [`terraform fmt`](/terraform/cli/commands/fmt) command automatically rewrites Terraform
```

  configuration files to a canonical format and style, so you don't have to

>   waste time making minor adjustments for readability and consistency. `fmt` works

Reviewer 2: ```suggestion
  waste time making minor adjustments for readability and consistency. The `terraform fmt` command works
```

  well as a pre-commit hook in your version control system.

> - [`terraform validate`](/terraform/cli/commands/validate): Validate the

Reviewer 2: ```suggestion
- The [`terraform validate`](/terraform/cli/commands/validate) commands validates the
```

  syntax and arguments of the Terraform configuration files in a directory,
  including argument and attribute names and types for resources and modules.
  The `plan` and `apply` commands automatically validate a configuration before

>   performing any other work. `validate` isn't a crucial part of the core

Reviewer 2: ```suggestion
  performing any other work. Running `validate` isn't a crucial part of the core
```

  workflow but can be very useful as a pre-commit hook or as part of a
  continuous integration pipeline.

> - [`0.13upgrade`](/terraform/cli/commands/0.13upgrade) and

Reviewer 2: ```suggestion
-  The [`0.13upgrade`](/terraform/cli/commands/0.13upgrade) and
```

>   [`0.12upgrade`](/terraform/cli/commands/0.12upgrade): Modify the configuration files in a Terraform module automatically to help deal with major

Reviewer 2: ```suggestion
  [`0.12upgrade`](/terraform/cli/commands/0.12upgrade) commands modify the configuration files in a Terraform module automatically to help deal with major
```

  syntax changes that occurred in the 0.13 and 0.12 releases of Terraform. Both
  commands are only available in the Terraform version they are
  associated with. Make sure to upgrade older code to be compatible
  with 0.12 before attempting to make it compatible with 0.13. For more detailed

>   information about updating code for new Terraform versions, see the [upgrade

Reviewer 2: ```suggestion
  information about updating code for new Terraform versions, refer to the [upgrade
```
